instrumentation/google.golang.org/grpc/otelgrpc/interceptor.go

```diff
@@ -211,19 +211,25 @@ func wrapClientStream(s grpc.ClientStream, desc *grpc.StreamDesc) *clientStream
 		// Both streams have to be closed
 		state := byte(0)
 
-		for event := range events {
-			switch event.Type {
-			case closeEvent:
-				state |= clientClosedState
-			case receiveEndEvent:
-				state |= receiveEndedState
-			case errorEvent:
-				finished <- event.Err
-				return
-			}
-
-			if state == clientClosedState|receiveEndedState {
-				finished <- nil
+		for {
+			select {
+			case event := <- events:
+				switch event.Type {
+				case closeEvent:
+					state |= clientClosedState
+				case receiveEndEvent:
+					state |= receiveEndedState
+				case errorEvent:
+					finished <- event.Err
+					return
+				}
+				if state == clientClosedState|receiveEndedState {
+					finished <- nil
+					return
+				}
+
+			case <-s.Context().Done():
```

Comment:
Disclaimer: I'm not a maintainer on this project but I have been looking at this code recently.

If we trust the ClientStream.Context() docs, it's not safe to use the Context() call here:
```
	// Context returns the context for this stream.
	//
	// It should not be called until after Header or RecvMsg has returned. Once
	// called, subsequent client-side retries are disabled.
	Context() context.Context
```

The alternative is to pass the parent context and check that.

Thanks for fixing this, I noticed this while fixing the other issue and this was next on my plate.